### ISSUE TITLE
Ability to auto assign permissions based on selected role

### ISSUE THREAD

**Issue by royjohngless** [KillahPotatoes/KP-Liberation#826]
### Is your feature request related to a problem? Please describe:
No it's not related to a problem. Would just be nice to be able to have permissions assigned based on the role selected when joining the mission so they don't have to manually be changed if someone changes roles. 

### Describe the solution you'd like:
Maybe an option like the ones in mission_params.hpp. Not a coder at all, just copying what I see there:  

class ManuallySetPermissions {
        title = $MANUALLY_SET_PERMS;
        values[] = {1, 0};
        texts[] = {$STR_PARAMS_ENABLED, $STR_PARAMS_DISABLED};
        default = 1;
};

So in that case it would be true and the permission system would function like it does now. If you set it to 0 you would have to edit values somewhere to define which role has which permissions. For example:

Role=CAS Pilot // Light Vehicles=false // Heavy Vehicles=false // Air Vehicles=true // Construction=false // Recycle=false // Others=true

### Additional context:

Our Liberation server has the revive settings so that any player can revive a downed teammate with a first aid kit. It kind of made the Medevac roles irrelevant. We renamed the roles in mission.sqm to Transport Pilot / Transport Pilot / Transport Engineer / Transport Engineer & changed the former medic traits to engineers so they could use toolkits to repair and defuse. We decided to have dedicated transport and disabled HALO and Mobile Respawns. Sometimes we will have players do Logistics with a chopper and go pick up supplies from factories. Eventually they get bored and switch to a ground team role and go attack sectors with others. They die and then build and fly a chopper out to the sector by themselves and just leave it there instead of needing one of the transport pilots or engineers to fly or drive them out. If I see that happen I pull their permissions for build and flight. Eventually they get back on again to do logistics and need those permissions back. It takes a lot of moderation on our end. Would be nice if all they had to do was a quick abort to role selection to automatically get the permissions for the role they are trying to play.

**Comment by Wyqer** [comment]
Will be considered for 0.96.8, but not for 0.96.7a anymore.

**Comment by royjohngless** [comment]
No worries, I know you have your hands full. Thanks for providing this mission, I'm a big fan. Beyond ecstatic you're even considering it, thanks again!

**Comment by Wyqer** [comment]
Thinking about extending the permissions parameter with another option, so you've
* Enabled
* Slot based
* Disabled

Would just make it necessary to "keep" the current recommendation of the troop organization concerning 2 squads, medevac, logistic, etc. if someone wants to use a slot based system. Or has more to tweak for individual changes, of course.

**Comment by royjohngless** [comment]
Makes sense. Well I leave it to the professional. Looking forward to these updates, they look awesome!
